### ISSUE TITLE
Continuing discussion about Magisk patch

### ISSUE THREAD

**Issue by ubergeek77** [ubergeek77/rattlesnakeos_scripts#1]
Hi @fltcaptriker,

The issue over on CaseyBakey's repository is getting rather large, and we're talking about very RattlesnakeOS-specific stuff now, so let's continue the discussion here.

Please be sure to read everything in my ReadMe before using this. I also recommend reading the comments in my `0001-patch_core_makefile.patch` so that you know what's going on, and which assumptions are being made.

As I said in the ReadMe, line 21 of the patch has not been tested, but since you probably aren't building a custom kernel, I've added it for your convenience.

Let me know if you run into any issues.

**Comment by fltcaptriker** [comment]
@ubergeek77 thanks, I'll over it carefully before I attempt to build. If I understand the readme correctly I should build the rom without the magisk script and flash it and then add the script and use OTA update?

**Comment by ubergeek77** [comment]
Yes, that's correct. However you need to do it, just make sure you're installing Magisk via an OTA update.

If you'd like to lock your bootloader too, then you should follow [dan-v's guide on that](https://github.com/dan-v/rattlesnakeos-stack/blob/10.0/FLASHING.md) when doing your initial install *without* Magisk, but *skip* his section in that guide titled "Disable OEM unlocking."

You should absolutely keep OEM unlocking *enabled* just so you can ensure nothing goes wrong.

Then, after locking the bootloader, start another RattlesnakeOS build with Magisk included. What's really cool is that you don't need to do anything special to install it - RattlesnakeOS' own background OTA updater will automatically start downloading the OTA, and it will install it for you. Then, when you reboot, you will have Magisk.

Once you have Magisk installed, you can leave the Magisk-adding script to your build, and it will automatically install the latest stable version of Magisk, and you *do not* (in fact, *should not*) need to follow [TopJohnWu's guide on installing OTA updates](https://github.com/topjohnwu/Magisk/blob/master/docs/tutorials.md#ota-installation). Just let RattlesnakeOS' OTA updater download the update, *don't* try to uninstall Magisk like in TopJohnWu's guide, and you will not only keep Magisk, it will be updated if there was a new version available.

I forgot to mention - my script does not add the Magisk Manager .apk. You will need to download and install that yourself, but when you do, you can open it and see that Magisk is installed. I'll add a note about that to the ReadMe.

**Comment by khimaros** [comment]
@ubergeek77 -- thank you for putting this together and for the detailed usage instructions. Regarding the factory reset limitation, is this something that could be improved with changes to the patch, or is it a limitation of Magisk itself?

**Comment by ubergeek77** [comment]
@khimaros Unfortunately, nothing can be done on our end to fix this.

This is a limitation of Magisk. More specifically, it is a consequence of the security features implemented in the closed-source Boot Control HAL. ***If*** this could be fixed, this would have to be addressed via a workaround in Magisk, and TopJohnWu doesn't seem interested in investigating it.

See:

https://github.com/topjohnwu/Magisk/issues, issue number 2214
